src/lib/Components/StickyTabPage/SnappyHorizontalRail.tsx

```diff
@@ -1,29 +1,44 @@
-import { useScreenDimensions } from "lib/utils/useScreenDimensions"
-import React, { useEffect } from "react"
+import React, { useImperativeHandle, useRef } from "react"
 import Animated from "react-native-reanimated"
 import { useAnimatedValue } from "./reanimatedHelpers"
 
-export const SnappyHorizontalRail: React.FC<{ offset: number }> = ({ offset, children }) => {
-  const currentOffset = useAnimatedValue(-offset)
-  const { width } = useScreenDimensions()
+export interface SnappyHorizontalRail {
+  setOffset(offset: number): void
+}
+export const SnappyHorizontalRail = React.forwardRef<
+  SnappyHorizontalRail,
+  React.PropsWithChildren<{ initialOffset?: number; width: number }>
+>(({ children, initialOffset = 0, width }, ref) => {
+  const currentOffset = useAnimatedValue(-initialOffset)
+  const currentAnimation = useRef<Animated.BackwardCompatibleWrapper>()
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      setOffset(offset) {
+        if (currentAnimation.current) {
+          currentAnimation.current.stop()
+        }
+        currentAnimation.current = Animated.spring(currentOffset, {
+          ...Animated.SpringUtils.makeDefaultConfig(),
+          stiffness: 600,
+          damping: 120,
+          toValue: -offset,
+        })
```

Comment:
Changed this to be controlled imperatively to avoid the react render cycle and give users *instant* feedback.